### PR TITLE
tlv: generate types for gossip unsigned range

### PR DESCRIPTION
**Pull request by ellemouton**
The gossip 1.75 messages will include a TLV range that the signatures will explicitly not cover. Generate a few types in that range (which will start from 160 as is done for bolt 12 messages). We will use this range for the actual signature fields in the new messages and later on we can use this range for SPV proofs and outpoint fields.

### CODE REVIEW

File: tlv/internal/gen/gen_tlv_types.go [hunk 1]
```diff
@@ -12,8 +12,13 @@ const (
 
 	// customTypeStart defines the beginning of the custom TLV type range as
 	// defined in BOLT-01.
-	customTypeStart   uint32 = 65536
-	defaultOutputFile        = "tlv_types_generated.go"
+	customTypeStart uint32 = 65536
+
+	// gossipUnsignedTypeStart defines the beginning of TLV range used in
+	// Gossip 1.75 messages that will not be covered by signatures.
+	gossipUnsignedTypeStart uint32 = 160
```
Comment (guggero) [inline tlv/internal/gen/gen_tlv_types.go:19]: Should we instead just increase `numberOfTypes` to 300 or something to that effect?

Comment (ellemouton) [reply tlv/internal/gen/gen_tlv_types.go:19]: ja could do! also need to add 2 marker types (one for 1000000000 and one for 3000000000). 

will update soon

**Review by guggero — APPROVED** [review]

File: tlv/internal/gen/gen_tlv_types.go (hunk 1 above)
Comment (ellemouton) [reply tlv/internal/gen/gen_tlv_types.go]: ah - realised i dont need the markers. So just did your suggestion and bumped the num gen to 300 :)

File: tlv/internal/gen/gen_tlv_types.go [hunk 2]
```diff
@@ -8,14 +8,28 @@ import (
...
 	// customTypeStart defines the beginning of the custom TLV type range as
 	// defined in BOLT-01.
-	customTypeStart   uint32 = 65536
-	defaultOutputFile        = "tlv_types_generated.go"
+	customTypeStart uint32 = 65536
+
+	// pureTLVSecondSignedTypeRangeStart defines the first TLV type of the
+	// second defined signed TLV range used in pure TLV messages.
+	pureTLVSecondSignedTypeRangeStart uint32 = 1000000000
+	// pureTLVSecondUnsignedTypeRangeStart defines the first TLV type of the
```
Comment (yyforyongyu) [inline tlv/internal/gen/gen_tlv_types.go:20]: missing newline here

File: tlv/tlv_types_generated.go [hunk 3]
```diff
@@ -1012,1012 +1012,5032 @@ func (t *tlvType100) tlv() {}
 
 type TlvType100 = *tlvType100
 
+type tlvType101 struct{}
```
Comment (yyforyongyu) [inline tlv/tlv_types_generated.go:1015]: Q: didn't follow the tlv package for a while...but I think we should only have two new types added here? why there are way more types here🤔

Comment (ellemouton) [reply tlv/tlv_types_generated.go:1015]: see the `numberOfTypes uint32 = 300` line :) 

bumped from 100 -> 300 so as to include 160 which is the start of the un-signed range

**Review by yyforyongyu — APPROVED** [review]
LGTM 🙏

File: tlv/tlv_types_generated.go (hunk 3 above)
Comment (Roasbeef) [reply tlv/tlv_types_generated.go:1015]: FWIW, you can always just make targeted types. They just need to be in this package. You could make a like `custom_types.go` file, then define like: 
```
type TlvType50001 = *tlvType50001

etc
```

**Pull request merged**